### DISCUSSION TITLE
bluetooth esp32 S3 box3

### DISCUSSION THREAD

**Discussion by youkorr** [BigBobbas/ESP32-S3-Box3-Custom-ESPHome#24] [Ideas]
hello is it possible to activate bluetooth to pair a bluetooth speaker??

**Comment by BigBobbas** [comment]
Hi , there is no support currently in ESPHome for bluetooth speakers, however I know that this is being worked on. I have tested bluetooth to use as a proxy for sensors etc. but this uses too much resource and causes the box to have issues.

**Reply by youkorr** [reply]
Ok but when it is possible it would be interesting to integrate it

**Reply by BigBobbas** [reply]
yes, definitely!

**Comment by schlick7** [comment]
On the bluetooth topic,
I just bought a switchbot curtain3 which from my understanding is controlled via bluetooth. I'm assuming I'd need the bluetooth proxy activated for this to be able to control the switchbot. Is bluetooth proxy still causing issues?

**Reply by BigBobbas** [reply]
Hi , I've not tried recently but it is documented in esphome docs that it will most likely cause issues with voice and a display.

**Reply by BigBobbas** [reply]
> On the bluetooth topic, I just bought a switchbot curtain3 which from my understanding is controlled via bluetooth. I'm assuming I'd need the bluetooth proxy activated for this to be able to control the switchbot. Is bluetooth proxy still causing issues?

just to confirm, i gave this a test today and adding BT freezes up the device , so unfortunately it is not going to be possible.

**Reply by schlick7** [reply]
Ah, that's a bummer. Thanks for checking it out

**Reply by schlick7** [reply]
So I got my box3 today and tried Bluetooth and it works a little, but then freezes eventually. Sometimes on the first voice try and sometimes it can get 1 or 2 commands out. 
I asked Nabu to close my curtains and it sent the bluetooth signal properly and then froze. 

This makes me think that it is actually pretty close to working. Not sure of the options for trying to slightly reduce RAM.

**Reply by BigBobbas** [reply]
possibly could remove the on device wake word stuff to free up RAM although the s3box does have 16mb PSRAM which is massive in comparison to other ESP32 S3 boards which usually have a max of 8mb.
 But to be honest for the cost of something like an esp32 s3 zero you could have one running as a proxy and easily fit it in the base of the dock and power it from the dock too most likely ;)

**Reply by schlick7** [reply]
Hadn't thought about just embedding a 2nd eps32 in the dock, that's a good idea. 

You don't think its a RAM issue then? just something isn't playing right? I'd like to play with this for a couple days, but I'm no developer. I did see that espressif says you can disable the classic Bluetooth(or BLE) stack, adjust heap/stack sizes in general, and even have Bluetooth use the flash storage for its heap/stack.  but I'm unsure if those go in the yaml or if they need to be in the original firmware. 

https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-guides/performance/ram-usage.html
https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/kconfig.html#config-bt-btc-task-stack-size
https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/bluetooth/controller_vhci.html
https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/kconfig.html#config-bt-ble-dynamic-env-memory

**Comment by BigBobbas** [comment]
you can see what PSRAM is free etc by adding in the debug component
https://esphome.io/components/debug.html#debug-component
But it is a known issue with voice and bluetooth https://esphome.io/components/voice_assistant.html see the red block at the top... this has been tried and tested hence the warning. so it is unlikely to get this to work consistently.  but obviously if you want to give it a go then that's your call :) 
TBH with regards to the espressif stuff I wouldn't know where to begin, I leave that to the devs ;)
